### ISSUE TITLE
You are not permitted to access this resource

### ISSUE THREAD

**Issue by stoleas** [tomasbjerre/pull-request-notifier-for-bitbucket#119]
Hi,

We are using Stash version v3.11.2 and v1.37 for Pull Request Notifier. We are having some users complain about inconsistency on when the plugin in triggered and it sometimes works but not always. I've tracked down an error that shows up around the same time as when the user reported issues and am seeing this stack trace in atlassian-stash.log 

```2016-05-25 23:16:27,797 ERROR [threadpool:thread-62]  c.a.s.i.c.StateTransferringExecutor Error while processing asynchronous task
com.atlassian.stash.exception.AuthorisationException: You are not permitted to access this resource
        at com.atlassian.stash.internal.aop.ExceptionRewriteAdvice.afterThrowing(ExceptionRewriteAdvice.java:36) ~[stash-platform-3.11.2.jar:na]
        at se.bjurr.prnfs.listener.PrnfsRenderer$PrnfsVariable.cloneUrlFromRepository(PrnfsRenderer.java:378) ~[na:na]
        at se.bjurr.prnfs.listener.PrnfsRenderer$PrnfsVariable.access$000(PrnfsRenderer.java:53) ~[na:na]
        at se.bjurr.prnfs.listener.PrnfsRenderer$PrnfsVariable$9.resolve(PrnfsRenderer.java:116) ~[na:na]
        at se.bjurr.prnfs.listener.PrnfsRenderer$PrnfsVariable.resolve(PrnfsRenderer.java:394) ~[na:na]
        at se.bjurr.prnfs.listener.PrnfsRenderer.render(PrnfsRenderer.java:446) ~[na:na]
        at se.bjurr.prnfs.listener.PrnfsPullRequestEventListener.notify(PrnfsPullRequestEventListener.java:186) ~[na:na]
        at se.bjurr.prnfs.listener.PrnfsPullRequestEventListener.handleEvent(PrnfsPullRequestEventListener.java:170) ~[na:na]
        at se.bjurr.prnfs.listener.PrnfsPullRequestEventListener$2.run(PrnfsPullRequestEventListener.java:138) ~[na:na]
        at com.atlassian.stash.internal.concurrent.StateTransferringExecutor$StateTransferringRunnable.run(StateTransferringExecutor.java:73) ~[stash-platform-3.11.2.jar:na]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_60]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_60]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_60]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_60]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_60]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60]
        ... 36 frames trimmed
Caused by: org.springframework.security.access.AccessDeniedException: Access is denied
        at org.springframework.security.access.vote.AffirmativeBased.decide(AffirmativeBased.java:83) ~[spring-security-core-3.2.7.RELEASE.jar:3.2.7.RELEASE]
        ... 16 common frames omitted```


Under the plugin configuration I have configured the plugin with 

"System Administrators" > Admin Restrictions
"Trigger PR Notifier Build" > Everyone > Trigger Button

**Comment by tomasbjerre** [comment]
So for the same user and button, it sometimes works and sometimes dont?

**Comment by tomasbjerre** [comment]
A workaround may be to use:

* ${PULL_REQUEST_TO_REPO_PROJECT_KEY} Example: PROJECT_1
* ${PULL_REQUEST_TO_REPO_SLUG} Example: rep_1

In the URL instead, to construct the clone URL.

**Comment by stoleas** [comment]
The user is saying that the button works 100% of the time. The filter I'm using 

Filter String > ${PULL_REQUEST_TO_REPO_NAME}
Filter regexp > reponame

Are you suggesting that I replace fitler string with "${PULL_REQUEST_TO_REPO_SLUG}"?

**Comment by tomasbjerre** [comment]
It looks like you are using ${PULL_REQUEST_FROM_SSH_CLONE_URL} in your post content field. It should work, but it looks like you have found a bug here.

So a workaround is to construct the clone URL using other variables. It may be possible with PULL_REQUEST_TO_REPO_PROJECT_KEY and PULL_REQUEST_TO_REPO_SLUG . But I'll have a closer look at solving the core problem as soon as I can.

**Comment by tomasbjerre** [comment]
I made a release in 1.38, does that solve your problem?

**Comment by stoleas** [comment]
I've upgraded the plugin, will get back to you once the user tests. Thanks.

**Comment by tomasbjerre** [comment]
"I'm not seeing the error anymore" do you mean that it is solved in 1.38?
"and it still seems to still be a problem" do you mean that it is not solved in 1.38?
